File: praxia_mobile_app/lib/ui/screens/re_test_trigger_screen.dart
import 'package:flutter/material.dart';
import '../widgets/app_drawer.dart';

class ReTestTriggerScreen extends StatelessWidget {
  const ReTestTriggerScreen({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: const Color(0xFFF8FAFC),
      appBar: AppBar(
        title: const Text('Re-Test Trigger', style: TextStyle(color: Colors.black87, fontWeight: FontWeight.bold)),
        backgroundColor: Colors.white,
        elevation: 0,
        leading: Builder(
          builder: (context) => IconButton(
            icon: const Icon(Icons.menu, color: Color(0xFF1D3B5A)),
            onPressed: () => Scaffold.of(context).openDrawer(),
          ),
        ),
      ),
      drawer: const AppDrawer(),
      body: SafeArea(
        child: Padding(
          padding: const EdgeInsets.all(20),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.center,
            children: [
              const SizedBox(height: 20),
              Container(
                width: 100,
                height: 100,
                decoration: BoxDecoration(color: const Color(0xFFEFF6FF), shape: BoxShape.circle),
                child: const Icon(Icons.edit_calendar_rounded, size: 50, color: Color(0xFF2563EB)),
              ),
              const SizedBox(height: 24),
              const Text('Next test recommended', style: TextStyle(fontSize: 16, color: Color(0xFF475569))),
              const SizedBox(height: 8),
              const Text('in 18 days', style: TextStyle(fontSize: 28, fontWeight: FontWeight.w900, color: Color(0xFF1E3A8A))),
              const SizedBox(height: 8),
              const Text('June 2, 2024', style: TextStyle(fontSize: 14, color: Color(0xFF64748B), fontWeight: FontWeight.w600)),
              const SizedBox(height: 40),

              _buildInfoCard(Icons.trending_up, 'Why retest?', 'To measure your progress and adjust your plan'),
              const SizedBox(height: 16),
              _buildInfoCard(Icons.science, "What we'll check", 'Track key biomarkers to see improvement'),

              const Spacer(),
              SizedBox(
                width: double.infinity,
                height: 54,
                child: ElevatedButton(
                  onPressed: () {
                    ScaffoldMessenger.of(context).showSnackBar(const SnackBar(content: Text('Test booked successfully!')));
                    Navigator.popUntil(context, (route) => route.isFirst);
                  },
                  style: ElevatedButton.styleFrom(backgroundColor: const Color(0xFF1E3A8A), shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(14))),
                  child: const Text('Book Follow-Up Test', style: TextStyle(color: Colors.white, fontWeight: FontWeight.bold, fontSize: 16)),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }

  Widget _buildInfoCard(IconData icon, String title, String subtitle) {
    return Container(
      padding: const EdgeInsets.all(16),
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.circular(16),
        boxShadow: [BoxShadow(color: Colors.black.withOpacity(0.04), blurRadius: 10, offset: const Offset(0, 4))],
      ),
      child: Row(
        children: [
          Icon(icon, color: const Color(0xFF2563EB), size: 28),
          const SizedBox(width: 16),
          Expanded(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(title, style: const TextStyle(fontWeight: FontWeight.bold, fontSize: 16, color: Color(0xFF1E293B))),
                const SizedBox(height: 4),
                Text(subtitle, style: const TextStyle(fontSize: 13, color: Color(0xFF64748B))),
              ],
            ),
          )
        ],
      ),
    );
  }
}

File: praxia_expo_app/src/screens/ReTestTriggerScreen.js
import React from 'react';
import { View, Text, StyleSheet, TouchableOpacity, SafeAreaView, Platform, StatusBar, Alert } from 'react-native';
import { Feather, MaterialIcons } from '@expo/vector-icons';
import { AppColors } from '../constants/theme';
import AppSidebarWrapper from '../components/AppSidebarWrapper';

export default function ReTestTriggerScreen({ navigation }) {
  const sidebarRef = React.useRef(null);
  return (
    <AppSidebarWrapper ref={sidebarRef} navigation={navigation}>
      <SafeAreaView style={styles.container}>
        <View style={styles.header}>
          <TouchableOpacity onPress={() => sidebarRef.current?.toggleDrawer()} style={styles.backButton}>
            <MaterialIcons name="menu" size={24} color="#1D3B5A" />
          </TouchableOpacity>
          <Text style={styles.headerTitle}>Re-Test Trigger</Text>
          <View style={{ width: 28 }} />
        </View>

      <View style={styles.content}>
        <View style={styles.centerSection}>
          <View style={styles.iconCircle}>
            <MaterialIcons name="edit-calendar" size={50} color="#2563EB" />
          </View>
          <Text style={styles.preTitle}>Next test recommended</Text>
          <Text style={styles.title}>in 18 days</Text>
          <Text style={styles.date}>June 2, 2024</Text>
        </View>

        <View style={styles.cardsContainer}>
          <InfoCard icon="trending-up" title="Why retest?" subtitle="To measure your progress and adjust your plan" />
          <InfoCard icon="science" title="What we'll check" subtitle="Track key biomarkers to see improvement" />
        </View>

        <View style={{ flex: 1 }} />

        <TouchableOpacity 
          style={styles.primaryButton}
          onPress={() => {
            Alert.alert("Success", "Test booked successfully!", [
              { text: "OK", onPress: () => navigation.navigate("Dashboard") }
            ]);
          }}
        >
          <Text style={styles.primaryButtonText}>Book Follow-Up Test</Text>
        </TouchableOpacity>
      </View>
      </SafeAreaView>
    </AppSidebarWrapper>
  );
}

const InfoCard = ({ icon, title, subtitle }) => (
  <View style={styles.card}>
    <MaterialIcons name={icon} size={28} color="#2563EB" />
    <View style={styles.cardTextContainer}>
      <Text style={styles.cardTitle}>{title}</Text>
      <Text style={styles.cardSubtitle}>{subtitle}</Text>
    </View>
  </View>
);

const styles = StyleSheet.create({
  container: { flex: 1, backgroundColor: '#F8FAFC', paddingTop: Platform.OS === 'android' ? StatusBar.currentHeight : 0 },
  header: { flexDirection: 'row', alignItems: 'center', justifyContent: 'space-between', paddingHorizontal: 16, paddingVertical: 15, backgroundColor: '#FFF' },
  backButton: { padding: 4 },
  headerTitle: { fontSize: 18, fontWeight: '700', color: AppColors.text },
  content: { padding: 20, flex: 1 },
  centerSection: { alignItems: 'center', marginTop: 20, marginBottom: 40 },
  iconCircle: { width: 100, height: 100, borderRadius: 50, backgroundColor: '#EFF6FF', justifyContent: 'center', alignItems: 'center', marginBottom: 24 },
  preTitle: { fontSize: 16, color: '#475569', marginBottom: 8 },
  title: { fontSize: 28, fontWeight: '900', color: '#1E3A8A', marginBottom: 8 },
  date: { fontSize: 14, fontWeight: '600', color: '#64748B' },
  cardsContainer: { marginBottom: 24 },
  card: { flexDirection: 'row', backgroundColor: '#FFF', borderRadius: 16, padding: 16, marginBottom: 16, shadowColor: '#000', shadowOffset: { width: 0, height: 2 }, shadowOpacity: 0.05, shadowRadius: 4, elevation: 2, alignItems: 'center' },
  cardTextContainer: { marginLeft: 16, flex: 1 },
  cardTitle: { fontSize: 16, fontWeight: 'bold', color: '#1E293B', marginBottom: 4 },
  cardSubtitle: { fontSize: 13, color: '#64748B' },
  primaryButton: { backgroundColor: '#1E3A8A', borderRadius: 14, height: 54, justifyContent: 'center', alignItems: 'center' },
  primaryButtonText: { color: 'white', fontSize: 16, fontWeight: 'bold' },
});
